### PR TITLE
don't loop forever in DHT poll

### LINKED ISSUE #2820: [LIBP2P] - Fix `DHTBehaviour::poll` spinning while loops
### What is this task and why do we need to work on it?

Analysing flame graphs indicated that there is an issue in `DHTBehaviour::poll`. There are two while loops that potentially block the execution.

### What work will need to be done to complete this task?

A similar issue was found in `DMBehaviour::poll` and was solved by a quick fix in [this PR](https://github.com/EspressoSystems/HotShot/pull/2769). Similar quick fix can be applied here.

### Are there any other details to include?

_No response_

### What are the acceptance criteria to close this issue?

`DHTBehaviour::poll` should not spin in the two while loops.

### Branch work will be merged to (if not the default branch)

_No response_

### PR DESCRIPTION
**Pull request by bfish713**
Closes #2820 

### This PR: 

Just loop over each request once in poll

### CODE REVIEW

File: crates/libp2p-networking/src/network/def.rs [hunk 1]
```diff
@@ -144,25 +143,13 @@ impl NetworkDef {
...
     /// Add a direct response for a channel
     pub fn add_direct_response(&mut self, chan: ResponseChannel<Vec<u8>>, msg: Vec<u8>) {
-        self.direct_message.add_direct_response(chan, msg);
-    }
-}
-
-impl From<DMEvent> for NetworkEventInternal {
-    fn from(event: DMEvent) -> Self {
-        Self::DMEvent(event)
+        let _ = self.direct_message.send_response(chan, msg);
```
Comment (jparr721) [inline crates/libp2p-networking/src/network/def.rs:152]: What is the `let _` being used for here?

**Review by jparr721 — COMMENTED** [review]
LGTM, but I'm less familiar with the previous behavior. One question

File: crates/libp2p-networking/src/network/def.rs (hunk 1 above)
Comment (bfish713) [reply crates/libp2p-networking/src/network/def.rs:152]: It's suppressing an error (and clipply).  I can log the error, but this is from the other PR

Comment (bfish713) [reply crates/libp2p-networking/src/network/def.rs:152]: tbh I'm not sure why this shows up even though this PR is to merge into the other branch

**Review by rob-maron — DISMISSED** [review]

**Pull request merged**